### ISSUE TITLE
code clean and optimize

### ISSUE THREAD

**Issue by yetist** [crosswire/xiphos#894]
Maybe we should clean up and optimize  the code if have enough time? I found those code should be cleaned up:
1. #if 0 ...... #endif
2. The commented or invalided code.
3. Deprecations api migrate, like gconf to gsettings, dbus-glib to gdbus, etc.
4. Use gresource to include resource files, like images, ui files, etc.
5. Rewrite real object to instead alias or macro defined object under src/xiphos_html.
6. Eliminate global and external variables, decouple code, and avoid the internal implementation of exposed objects.
7. The address  of the license is located in the header of some files.

**Comment by karlkleinpaste** [comment]
particularly for the "#if 0" case, that has at times been a habit of mine, to remove something from current use but keep it present (and visible, especially, so it's not forgotten) in case it needs to be brought back into use later. i recall using that a number of times some years ago, when doing a round of cleanups consequent to cppcheck analysis -- quite a few unused functions were kept within #if 0 blocks because i wasn't sure actual removal was a good idea. perhaps it's time to make them go away.

but otherwise, yes, it seems there is always a great deal of clean-up opportunity available. there is a lot of quite old code -- old elements of the settings structure, no longer used, for example. what was previously called gnomesword started in 2000, using gtk1, and dead artifacts from almost 2 decades ago remain.

**Comment by teusbenschop** [comment]
About the deprecated 'dbus-glib', I am adding information from Debian that includes a recommendation and a porting guide. I hope it helps.


Source: xiphos
Severity: normal
Tags: sid bullseye
Control: block 895291 by -1
User: pkg-utopia-maintainers@lists.alioth.debian.org
Usertags: dbus-glib-deprecation

dbus-glib is a deprecated D-Bus library with some significant design
flaws, and is essentially unmaintained. I would like to minimize its
use, and eventually remove it from Debian. There will not be a
version that fixes its design flaws, because that would be a major
compatibility break, and any user of dbus-glib who is willing to port
their application to a newer, incompatible version should instead be
porting their application to a better D-Bus implementation such as
GDBus.

For most purposes, the recommended replacement for dbus-glib is the
GDBus family of APIs in GLib, found in <gio/gio.h>. This does not add
an additional dependency, because dbus-glib already depends on a
sufficiently new version of GLib. A porting guide is available in the
GLib documentation:
<https://developer.gnome.org/gio/stable/ch35.html>. Practical
examples of porting from dbus-glib to GDBus can be found in the git
history of most older GNOME applications.

Alternatives to GDBus, with different design emphasis and trade-offs,
include sd-bus (systemd's D-Bus implementation), QtDBus (Qt's D-Bus
API), and libdbus (the low-level reference D-Bus implementation).
Please contact the D-Bus mailing list <dbus at lists.freedesktop.org>
if you are unsure which D-Bus implementation is most suitable for a
particular package.

Some libraries expose dbus-glib as part of their API/ABI, in which
case removing the deprecated dependency requires breaking API/ABI
(telepathy-glib is a good example). For these libraries, maintainers
should talk to the dependent library's upstream developers about
whether the dependent library should break API/ABI and switch to
GDBus, or whether the dependent library should itself be deprecated.

In a few cases, the package uses the reference D-Bus library libdbus
for all D-Bus-related APIs, and only uses dbus-glib as a way to
connect libdbus to the GLib main loop: if the only functions
referenced from dbus-glib are dbus_connection_setup_with_g_main() and
dbus_server_setup_with_g_main(), then you are in this situation. The
recommended replacement in this case is to bundle the dbus-gmain
branch from the dbus-glib git repository, for example as a `git
subtree` or `git submodule`. For example, dbus-python's GLib
integration now works like this. See
<https://gitlab.freedesktop.org/dbus/dbus-glib/blob/dbus-gmain/README
.md> for more details.

**Comment by stappersg** [comment]
> About the deprecated 'dbus-glib', 

That is about item 3, _Deprecations api migrate, like gconf to gsettings, dbus-glib to gdbus, etc._ from the opening of this issue.

Did it happen or did we learn that it is not wise to report several issues in a single issue?

**Comment by teusbenschop** [comment]
Yeah, it would be easier to keep track of migrating "dbus-glib to gdbus" in a separate issue. That's easier for the Xiphos upstream developers to keep track of this work.

Currently Xiphos is still in Debian. But is in danger of getting pushed out of Debian once the "dbus-glib" library is removed from Debian. So migrating this API looks like it's a important thing.

**Comment by teusbenschop** [comment]
I opened a new issue for migrating away from dbus-glib:

https://github.com/crosswire/xiphos/issues/1049

**Comment by LAfricain** [comment]
~~1. #if 0 ...... #endif~~
~~2. The commented or invalided code.~~
~~3. Deprecations api migrate, like gconf to gsettings, dbus-glib to gdbus, etc.~~
~~4. Use gresource to include resource files, like images, ui files, etc.~~
5. Rewrite real object to instead alias or macro defined object under src/xiphos_html.
6. Eliminate global and external variables, decouple code, and avoid the internal implementation of exposed objects.
~~7. The address of the license is located in the header of some files.~~

Items 2, 4, 7 done with #1381
